client/src/components/Clan/RoleAssignmentModal.tsx

```diff
@@ -0,0 +1,147 @@
+import { Box, Checkbox, FormControl, InputLabel, MenuItem, Paper, Select } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import FantasyButton from '../FantasyButton';
+import Text from '../Text';
+import { useAlert } from '../../hooks/useAlert';
+import Server from '../../utils/Server';
+import catchError from '../../utils/catchError';
+
+interface Member {
+  id: string;
+  name: string;
+}
+
+interface Role {
+  id: string;
+  name: string;
+}
+
+interface RoleAssignmentModalProps {
+  open: boolean;
+  onClose: () => void;
+  clanId: string;
+  role: Role | null;
+  members: Member[];
+  onAssigned: () => void;
+}
+
+const RoleAssignmentModal: React.FC<RoleAssignmentModalProps> = ({
+  open,
+  onClose,
+  clanId,
+  role,
+  members,
+  onAssigned,
+}) => {
+  const { t } = useTranslation();
+  const Alert = useAlert();
+
+  const [selectedMembers, setSelectedMembers] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (open) {
+      setSelectedMembers([]);
+    }
+  }, [open, members]);
+
+  const handleAssign = () => {
+    if (selectedMembers.length === 0 || !role) {
+      Alert.open('error', t('pleaseSelectMember'));
+      return;
+    }
+
+    // Assign role to all selected members
+    const promises = selectedMembers.map(
+      (memberId) => Server.Clan.assignRole(clanId, memberId, role.id),
+    );
+
+    Promise.all(promises)
+      .then(() => {
+        Alert.open('success', t('rolesAssigned', { count: selectedMembers.length }));
+        setSelectedMembers([]);
+        onAssigned();
+        onClose();
+      })
+      .catch(catchError(Alert));
+  };
+
+  if (!open || !role) return null;
+
+  return (
+    <Box
+      sx={{
+        position: 'fixed',
+        top: 0,
+        left: 0,
+        right: 0,
+        bottom: 0,
+        bgcolor: 'rgba(0, 0, 0, 0.8)',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        zIndex: 1400,
+      }}
+      onClick={onClose}
+    >
```

Comment:
I'm guessing this is the backdrop to the modal you built manually?
Since we're using MUI, it would be better to use their built-in solution instead:
https://mui.com/material-ui/react-modal/